.github/workflows/function-composition.yml

```diff
@@ -0,0 +1,354 @@
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Login to Docker Hub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+      - name: Set up Docker Buildx
+        id: buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Build and push
+        working-directory: ./function-images/tests/chained-function-serving
+        run: |
+          make all-image
+          make all-image-push
+
+  push-chained-functions-eventing:
+      name: Push chained-functions-eventing
+      needs: push-golang
+      runs-on: ubuntu-18.04
+      strategy:
+        fail-fast: false
+      steps:
+        - uses: actions/checkout@v2
+
+        - name: Login to Docker Hub
+          uses: docker/login-action@v1
+          with:
+            username: ${{ secrets.DOCKER_HUB_USERNAME }}
+            password: ${{ secrets.DOCKER_HUB_ACCESS_TOKEN }}
+
+        - name: Set up Docker Buildx
+          id: buildx
+          uses: docker/setup-buildx-action@v1
+
+        - name: Build and push
+          working-directory: ./function-images/tests/chained-function-eventing
+          run: make push-all
+
+  bench-docker-compose-serving:
+    name: Serving uBenchmark (docker-compose)
+    needs: push-chained-functions-serving
+    runs-on: ubuntu-18.04
+    strategy:
+      fail-fast: false
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          lfs: 'true'
+
+      - name: start multi-container program with docker-compose
+        working-directory: function-images/tests/chained-function-serving
+        run: |
+          docker-compose up &> log_file &
+          sleep 15s
+
+      - name: invoke the chain
+        run: |
+          ./bin/grpcurl -max-time 10 -plaintext localhost:3031 helloworld.Greeter.SayHello
+
+      - name: show docker-compose log
+        run: cat function-images/tests/chained-function-serving/log_file
+
+  bench-serving:
+    name: Serving uBenchmark (Knative)
+    needs: push-chained-functions-serving
+    env:
+        GITHUB_RUN_ID: ${{ github.run_id }}
+        LOGPATH: /tmp/uBench-logs/${{ github.run_id }}
+        GOCACHE: /root/tmp/gocache
+        GOPATH: /root/tmp/gopath
+        TMPDIR: /root/tmp
+    runs-on: [stock-knative]
+    strategy:
+      fail-fast: false
+    steps:
+      - name: Setup TMPDIR
+        run: mkdir -p $TMPDIR
+
+      - name: Set up Golang
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.16
+
+      - name: Setup LOGPATH
+        run: |
+          mkdir -p /tmp/uBench-logs/${{ github.run_id }}
+
+      - name: Host Info
+        run: |
+          echo $HOSTNAME
+          echo $GITHUB_RUN_ID
+          echo $LOGPATH
+
+      - uses: actions/checkout@v2
+        with:
+          lfs: 'true'
+
+      - name: Checkout LFS objects
+        run: git lfs checkout
+
+      - name: Pull latest images
```

Comment:
remove this step. it should not be necessary if clean-up is successful. please double check